maven-embedder/src/main/java/org/apache/maven/cli/internal/ExtensionResolutionException.java

```diff
@@ -0,0 +1,59 @@
+package org.apache.maven.cli.internal;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.maven.cli.internal.extension.model.CoreExtension;
+
+/**
+ * Exception occurring trying to resolve a plugin.
+ *
+ * @author <a href="mailto:brett@apache.org">Brett Porter</a>
+ */
+public class ExtensionResolutionException
+    extends Exception
+{
+
+    private final CoreExtension extension;
+
+    public ExtensionResolutionException( CoreExtension extension, Throwable cause )
+    {
+        super( "Extension " + getId( extension ) + " or one of its dependencies could not be resolved: "
+                        + cause.getMessage(), cause );
+        this.extension = extension;
+    }
+
+    public CoreExtension getExtension()
+    {
+        return extension;
+    }
+
+    public static String getId( CoreExtension extension )
+    {
+        StringBuilder id = new StringBuilder( 128 );
+
+        id.append( ( extension.getGroupId() == null ) ? "[unknown-group-id]" : extension.getGroupId() );
```

Comment:
Do we expect them to be be null??

Reply:
I don't see any place where this is checked, so my best guess is that if the user does not fill it, it will be null.

Reply:
I wonder what the resolver will say when the input in non-sense...

Reply:
I suppose an exception will be thrown.  Fwiw, the `getId()` is simply copied from the `Plugin` class and that's what is used in the `PluginResolutionException`.

Reply:
Then why not have it generated by Modello just like for `Plugin`?

Reply:
Fixed